assets/js/swap-phrase.js

```diff
@@ -0,0 +1,13 @@
+function init(field) {
```

Comment:
We can deal with it as a reusable component. Imagine that we could have other buttons on the screen that do the same thing even on another page. Would be nice if we could just replicate the markup and everything work as expected without changing any JS code.

We could achieve that writing a markup like that:

```HTML
<div data-component>
  <p data-phrase class="lead text-center">
   ...
  </p>

  <div class="text-center">
    <button type="button" data-button class="btn">
      Gerar outra frase
    </button>
  </div>
</div>
```

And the JS like that:
```JS
const fetchRandomPhrase = () => {
  fetch("/api/phrase")
    .then(response => response.json())
    .then(json => json.data)
}

const component = (element) => {
  // Find elements that the scrip will need from the component.
  // Notice that this approach avoid look up in the entire DOM.
  const phraseTarget = element.querySelector("data-phrase")
  const button = element.querySelector("[data-button]")
  
  button.addEventListener("click", event => {
    phrase = fetchRandomPhrase()

    phraseTarget.innerHTML = json.phrase
  });
}

// For each component that found, plug this JS.
document.querySelectorAll("[data-component]").forEach(element => component(element))
```

Other advantages that I see:
* We avoid look ups in the entire DOM, like you are doing here: `document.querySelector(event.target.dataset.phraseField);`
* The script is a little bit readable. You can see the elements that the script will use in the component construction.
* It's easy to replicate the markup case you need to write a test for it.

Wdyt?

Reply:
I tried to use your code but I couldn't make it work this way.

Look at this part of code:
```js
const fetchRandomPhrase = () => {
  fetch("/api/phrase")
    .then(response => response.json())
    .then(json => json.data)
}
```
It don't return an object or something like that. It returns a `Promise`, so I can't just fetch the phrase and use it like that:
```js
let phrase = fetchRandomPhrase()
phraseTarget.innerHTML = json.data
```

I had to return a Promise here:
```js
const fetchRandomPhrase = () => {
  return fetch("/api/phrase")
    .then(response => response.json())
}
```

And then use this promise to insert the phrase:
```js
button.addEventListener("click", event => {
    fetchRandomPhrase()
      .then(json => phraseTarget.innerHTML = json.data)
  });
```

Is it ok? 🤔

Reply:
yeah, it'is. It's good because you let to the code that is depending on the request to deal with it.

Reply:
Done